test/AElf.Runtime.CSharp.Tests.BadContract/AElf.Runtime.CSharp.Tests.BadContract.csproj

```diff
@@ -0,0 +1,40 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <AssemblyName>AElf.Runtime.CSharp.Tests.BadContract</AssemblyName>
+        <RootNamespace>AElf.Runtime.CSharp.Tests.BadContract</RootNamespace>
+    </PropertyGroup>
+    
+    <ItemGroup>
+      <ProjectReference Include="..\..\src\AElf.Sdk.CSharp\AElf.Sdk.CSharp.csproj" />
+    </ItemGroup>
+    
+    <ItemGroup>
+        <Content Include="..\..\protobuf\aelf_options.proto">
+            <Link>Protobuf\Proto\aelf_options.proto</Link>
+        </Content>
+        <Content Include="..\..\protobuf\common.proto">
+            <Link>Protobuf\Proto\common.proto</Link>
+        </Content>
+        <Content Include="..\..\protobuf\bad_contract.proto">
+            <Link>Protobuf\Proto\bad_contract.proto</Link>
+        </Content>
+    </ItemGroup>
+    
+    <Target Name="Gen_proto" BeforeTargets="BeforeBuild">
+        <Exec Condition="!$([MSBuild]::IsOsPlatform(Windows))" Command="../../scripts/generate_contract_code.sh bad_contract.proto" />
+        <Exec Condition="$([MSBuild]::IsOsPlatform(Windows))" Command="..\..\scripts\generate_contract_code.bat bad_contract.proto" />
+    </Target>
+    
+    <Target Name="UpdateGeneratedFiles" DependsOnTargets="Gen_proto" BeforeTargets="BeforeBuild">
+        <ItemGroup>
+            <Compile Include="./Protobuf/Generated/*.cs" />
+        </ItemGroup>
+    </Target>
+    
+    <Target Name="CopyContractDll" AfterTargets="AfterBuild">
+        <Exec Condition="!$([MSBuild]::IsOsPlatform(Windows))" Command="cp $(TargetDir)$(TargetName).dll $(SolutionDir)/test/AElf.Runtime.CSharp.Tests/contracts" />
```

Comment:
$(SolutionDir) will be *Undefined* if build with *.csproj

Reply:
Fixed by using relative path in the last commit.